### DISCUSSION TITLE
Yarp installation error

### DISCUSSION THREAD

**Discussion by Akashleena** [robotology/community#418] [Q&A] [answered]
I do have YCMConfig.cmake.in in this path /home/aleena/icubLearningTrajectories/icub/software/src/ycm/YCMConfig.cmake.in. I have yarp in this path /home/aleena/icubLearningTrajectories/icub/software/src/yarp . When I am trying to cmake in yarp I get this error

Could not find a package configuration file provided by "YCM" (requested
version 0.11.0) with any of the following names:

YCMConfig.cmake
ycm-config.cmake

Add the installation prefix of "YCM" to CMAKE_PREFIX_PATH or set "YCM_DIR"
to a directory containing one of the above files. If "YCM" provides a
separate development package or SDK, be sure it has been installed.

-- Configuring incomplete, errors occurred!

How do I resolve this error?

**Comment by PeterBowman** [comment, marked as answer]
Hi, @Akashleena. The ` YCMConfig.cmake.in` file is just a template used by CMake to generate the final ` YCMConfig.cmake` file inside the `build` directory of YCM. You should build this project first so that YARP can find it as part of its own build process. Namely:

```
cd /home/aleena/icubLearningTrajectories/icub/software/src/ycm
mkdir build
cd build
cmake ..
make
sudo make install # skip this if you don't have permissions or don't want to mess with system files
```

Now, `cmake` again into YARP. If YCM is not found yet, try `cmake -DYCM_DIR=/home/aleena/icubLearningTrajectories/icub/software/src/ycm/build`.

[a comment by an automated account is not shown]

**Comment by pattacini** [comment]
Closing, feel free to open it up again if needed.
